specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json

```diff
@@ -0,0 +1,257 @@
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Creates or updates the Defender for Storage settings on a specified resource.": {
+            "$ref": "./examples/DefenderForStorage/PutDefenderForStorageSettings_example.json"
+          }
+        },
+        "tags": [
+          "DefenderForStorage"
+        ],
+        "description": "Creates or updates the Defender for Storage settings on a specified storage account.",
+        "operationId": "DefenderForStorage_Create",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceId"
+          },
+          {
+            "$ref": "#/parameters/DefenderForStorageSettingName"
+          },
+          {
+            "$ref": "#/parameters/DefenderForStorageSetting"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create or update Defender for Storage settings.",
+            "schema": {
+              "$ref": "#/definitions/DefenderForStorageSetting"
+            }
+          },
+          "201": {
+            "description": "Request to  create or update Defender for Storage settings partially succeeded.",
+            "schema": {
+              "$ref": "#/definitions/DefenderForStorageSetting"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DefenderForStorageSetting": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DefenderForStorageSettingProperties"
+        }
+      },
+      "description": "The Defender for Storage resource."
+    },
+    "DefenderForStorageSettingProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether Defender for Storage is enabled on this storage account."
+        },
+        "malwareScanning": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/MalwareScanningProperties"
+        },
+        "sensitiveDataDiscovery": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SensitiveDataDiscoveryProperties"
+        },
+        "overrideSubscriptionLevelSettings": {
+          "type": "boolean",
```

Comment:
Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important

Reply:
See reply on the previous comment